edocumentation-service/src/main/java/org/dcsa/edocumentation/service/unofficial/BookingValidationService.java

```diff
@@ -27,6 +28,13 @@ public class BookingValidationService {
 
   private final static Set<BkgDocumentStatus> CAN_BE_VALIDATED =
     Set.of(BkgDocumentStatus.RECE, BkgDocumentStatus.PENC);
+  private static final Predicate<LocationType> HAS_SOURCE_LOCATION = code ->
+          code == LocationType.PRE || code == LocationType.POL;
+  private static final Predicate<LocationType> HAS_DESTINATION_LOCATION = code ->
+          code == LocationType.POD || code == LocationType.PDE;
+  private static final Predicate<ShipmentLocation> HAS_SOURCE_OR_DESTINATION_LOCATION =
+      sl -> HAS_SOURCE_LOCATION.test(sl.getShipmentLocationTypeCode())
+              || HAS_DESTINATION_LOCATION.test(sl.getShipmentLocationTypeCode());
```

Comment:
Nit: This is `Predicate.or`

Reply:
(Feel free to resolve and merge without changing; it is just a minor suggestion)